src/Symfony/Component/Validator/Constraints/UrlValidator.php

```diff
@@ -62,6 +62,18 @@ public function validate($value, Constraint $constraint)
             $this->buildViolation($constraint->message)
                 ->setParameter('{{ value }}', $this->formatValue($value))
                 ->addViolation();
+
+            return;
+        }
+
+        if ($constraint->checkDNS) {
+            $host = parse_url($value, PHP_URL_HOST);
+
+            if (!checkdnsrr($host, 'ANY')) {
+                $this->buildViolation($constraint->message)
```

Comment:
The message in this case should be different as the URL is valid. It just does not resolve to anything.